dapp/src/components/Collateral.js

```diff
@@ -0,0 +1,132 @@
+import React, { useState, useEffect } from 'react'
+import { Typography } from '@originprotocol/origin-storybook'
+import { assetRootPath } from 'utils/image'
+import { PieChart } from 'react-minimal-pie-chart'
+import { formatCurrency } from '../utils/math'
+import useCollateralQuery from '../queries/useCollateralQuery'
+import { useStoreState } from 'pullstate'
+import ContractStore from 'stores/ContractStore'
+
+const Collateral = () => {
+  const collateral = useStoreState(ContractStore, (s) => {
+    return s.collateral || {}
+  })
+
+  const collateralQuery = useCollateralQuery({
+    onSuccess: (collateral) => {
+      ContractStore.update((s) => {
+        s.collateral = collateral
+      })
+    },
+  })
+
+  useEffect(() => {
+    collateralQuery.refetch()
+  }, [])
+
+  const total = collateral.collateral?.reduce((t, s) => {
+    return { total: Number(t.total) + Number(s.total) }
+  }).total
+
+  const colors = {
```

Comment:
nit: Move constants out of the component

Reply:
done